app/src/main/java/com/concordia/mcga/adapters/POISearchAdapter.java

```diff
@@ -0,0 +1,281 @@
+     * @param viewGroup The parent that this view will be attached to
+     * @return The inflated POI group view
+     */
+    @Override
+    public View getGroupView(int groupPosition, boolean b, View view, ViewGroup viewGroup) {
+        Campus campus = (Campus)getGroup(groupPosition);
+
+        if (campus == null) {
+            return null;
+        }
+
+        if (view == null) {
+            LayoutInflater layoutInflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
+            view = layoutInflater.inflate(R.layout.poi_search_group_row, null);
+        }
+
+        CircleImageView campusImage = (CircleImageView) view.findViewById(R.id.groupImage);
+        Campus camp = (Campus)getGroup(groupPosition);
+        if (camp == Campus.SGW) {
+            campusImage.setImageResource(R.mipmap.ic_sgw_campus);
+        } else {
+            campusImage.setImageResource(R.mipmap.ic_loy_campus);
+        }
+
+        TextView campusRow = (TextView) view.findViewById(R.id.groupText);
+        campusRow.setText(campus.getName());
+
+        return view;
+    }
+
+    /**
+     * Inflates and returns the view associated with the POI list item. For now, a building name
+     * @param groupPosition The group/campus index
+     * @param childPosition The child/POI index
+     * @param b Whether the child is the last one in the group
+     * @param view Existing view to reuse if possible
+     * @param viewGroup The parent group this view will be attached to
+     * @return The POI menu item (building name)
+     */
+    @Override
+    public View getChildView(int groupPosition, int childPosition, boolean b, View view, ViewGroup viewGroup) {
+        Building building = (Building)getChild(groupPosition, childPosition);
+
+        if (building == null) {
+            return null;
+        }
+
+        if (view == null) {
+            LayoutInflater layoutInflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
+            view = layoutInflater.inflate(R.layout.poi_search_child_row, null);
+        }
+
+        TextView buildingRow = (TextView) view.findViewById(R.id.childText);
+        buildingRow.setText(building.getName());
+
+        return view;
+    }
+
+    @Override
+    public boolean isChildSelectable(int i, int i1) {
+        return true;
+    }
+
+    /**
+     * This is the actual filter to select the POIs specified by the passed string.
+     * The POIs that are determined to meet the query criteria are included in the filtered lists
+     * (see constructor.)
+     * @param query The string query to match POIs against
+     */
+    public void filterData(String query) {
+        query = query.toLowerCase();
+        sgwFilteredList.clear();
+        loyolaFilteredList.clear();
+
+        if (query.isEmpty()) {
+            //sgwFilteredList.addAll(Campus.SGW.getBuildings());
+            //loyolaFilteredList.addAll(Campus.LOY.getBuildings());
```

Comment:
I chose this as the functionality based on my experience testing it. The reason I left them commented is because I wanted to make it easy to switch to the showing all case in the event that's what we want